app/ante/eth.go

```diff
@@ -549,3 +549,25 @@ func (issd EthIncrementSenderSequenceDecorator) AnteHandle(ctx sdk.Context, tx s
 	// set the original gas meter
 	return next(ctx, tx, simulate)
 }
+
+// NOTICE: Adapted from ValidateBasicDecorator from cosmos-sdk, ignore ErrNoSignatures
+type EthValidateBasicDecorator struct{}
+
+func NewEthValidateBasicDecorator() EthValidateBasicDecorator {
+	return EthValidateBasicDecorator{}
+}
+
+func (vbd EthValidateBasicDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// no need to validate basic on recheck tx, call next antehandler
+	if ctx.IsReCheckTx() {
+		return next(ctx, tx, simulate)
+	}
+
+	err := tx.ValidateBasic()
```

Comment:
would it makes more sense to cast to MsgEthereumTx and call validateBasic on each messages?

```
	for i, msg := range tx.GetMsgs() {
		msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
		msgEthTx.ValidateBasic()
		
```


This will call the custom ValidateBasic defined there		
https://github.com/tharsis/ethermint/blob/main/x/evm/types/msg.go#L92

Reply:
seemingly, `msg.ValidateBasic` is called before ante handler: https://github.com/cosmos/cosmos-sdk/blob/master/baseapp/baseapp.go#L622.